File: zig/src/types/IndexEntry.zig
const std = @import("std");

pub const IndexEntry = struct {
    path: []const u8,
    sha: []const u8,
    mode: []const u8,
    size: u32,
    ctime_seconds: u32,
    ctime_nanos: u32,
    mtime_seconds: u32,
    mtime_nanos: u32,
    dev: u32,
    ino: u32,
    uid: u32,
    gid: u32,
};

File: dotnet/Gitologist/Types/IndexEntry.cs
namespace Gitologist.Types;

public record class IndexEntry
{
    public required string Path { get; init; }
    public required string Sha { get; init; }
    public required string Mode { get; init; }
    public required uint Size { get; init; }
    public required uint CtimeSeconds { get; init; }
    public required uint CtimeNanos { get; init; }
    public required uint MtimeSeconds { get; init; }
    public required uint MtimeNanos { get; init; }
    public required uint Dev { get; init; }
    public required uint Ino { get; init; }
    public required uint Uid { get; init; }
    public required uint Gid { get; init; }
}
